stdlib/public/Differentiation/DifferentiationSupport.swift

```diff
@@ -1262,7 +1262,36 @@ public extension Array where Element: Differentiable {
   }
 
   @usableFromInline
-  @derivative(of: differentiableReduce)
+  @derivative(of: differentiableReduce, wrt: (self, initialResult))
+  func _jvpDifferentiableReduce<Result: Differentiable>(
+    _ initialResult: Result,
+    _ nextPartialResult: @differentiable (Result, Element) -> Result
+  ) -> (value: Result,
+        differential: (Array.TangentVector, Result.TangentVector)
```

Comment:
minor:
```suggestion
        differential: (TangentVector, Result.TangentVector)
```

Reply:
I think keeping this is good for clarity. The VJP also keeps `Array.`. I don't feel strongly though.